s3.py

```diff
@@ -27,58 +27,77 @@ class S3Uploader(MirrorUploader):
 
     @classmethod
     def from_config(cls, _db):
+        """Create an S3Uploader from site configuration.
+
+        :return: An S3Uploader if S3 is configured; otherwise None.
+        """
         integration = cls.integration(_db)
-        cls.initialize_buckets(_db)
+        if not integration:
+            return None
+        cls.initialize_buckets(_db)        
```

Comment:
If there's no S3 integration (as is the case in the circ manager) then this should return None (indicating that there is no way to upload things to S3) rather than raising an error.